src/mlpack/bindings/cli/add_to_po.hpp

```diff
@@ -14,95 +13,171 @@
+{
+  app.add_option_function<T>(cliName.c_str(),
+      [&param](const T& value)
+      {
+        param.value = value;
+        param.wasPassed = true;
+      },
+      param.desc.c_str());
+}
+
+/**
+ * Add an option to CLI11.
+ * 
+ * @param cliName The name of the option to add to CLI11.
  * @param descr Description string for parameter.
  * @param desc Options description to add parameter to.
  */
 template<typename T>
-void AddToPO(const std::string& boostName,
-             const std::string& descr,
-             boost::program_options::options_description& desc,
-             const typename boost::enable_if<util::IsStdVector<T>>::type* = 0,
-             const typename boost::disable_if<std::is_same<T, bool>>::type* = 0)
+void AddToPO(const std::string& cliName,
+             util::ParamData& param,
+             CLI::App& app,
+             const typename boost::disable_if<std::is_same<T, bool>>::type* = 0,
+             const typename boost::disable_if<arma::is_arma_type<T>>::type* = 0,
+             const typename boost::disable_if<data::HasSerialize<T>>::type* = 0,
+             const typename boost::disable_if<std::is_same<T,
+                std::tuple<mlpack::data::DatasetInfo, arma::mat>>>::type* = 0)
 {
-  desc.add_options()(boostName.c_str(),
-      boost::program_options::value<T>()->multitoken(), descr.c_str());
+  app.add_option_function<T>(cliName.c_str(),
+      [&param](const T& value)
+      {
+        param.value = value;
+        param.wasPassed = true;
+      },
+      param.desc.c_str());
 }
 
 /**
- * Add a boolean option to boost::program_options.
+ * Add a boolean option to CLI11.
  *
- * @param boostName The name of the option to add to boost::program_options.
+ * @param cliName The name of the option to add to CLI11.
  * @param descr Description string for parameter.
  * @param desc Options description to add parameter to.
  */
 template<typename T>
-void AddToPO(const std::string& boostName,
-             const std::string& descr,
-             boost::program_options::options_description& desc,
-             const typename boost::disable_if<util::IsStdVector<T>>::type* = 0,
-             const typename boost::enable_if<std::is_same<T, bool>>::type* = 0)
+void AddToPO(const std::string& cliName,
+             util::ParamData& param,
+             CLI::App& app,
+             const typename boost::enable_if<std::is_same<T, bool>>::type* = 0,
+             const typename boost::disable_if<arma::is_arma_type<T>>::type* = 0,
+             const typename boost::disable_if<data::HasSerialize<T>>::type* = 0,
+             const typename boost::disable_if<std::is_same<T,
+                std::tuple<mlpack::data::DatasetInfo, arma::mat>>>::type* = 0)
 {
-  desc.add_options()(boostName.c_str(), descr.c_str());
+  app.add_flag_function(cliName.c_str(),
+      [&param](const T& value)
+      {
+        param.value = value;
+        param.wasPassed = true;
+      },
+      param.desc.c_str());
```

Comment:
This overload is fine---it applies when `T` is bool, for which this implementation is already correct.